### ISSUE TITLE
Incorrect `MA0040` with extension method to CancellationToken

### ISSUE THREAD

**Issue by vchirikov** [meziantou/Meziantou.Analyzer#228]
Hi, @meziantou.

**Context**
Rule identifier: _MA0040_
Target Framework: Any
Language version: Any

**Describe the bug**
I have an extension method to `CancellationToken` and call to it causes reporting of `MA0040 - Flow the cancellation token`

**To Reproduce**
```cs
public static class WaitExtensions 
{
  public static Task WaitAsync(this CancellationToken cancellationToken) => throw default;
}

public class MyCode
{ 
  void MyMethod() => new CancellationToken().WaitAsync(); // MA0040 
}

```
p.s. thanks for your analyzer :)

**Comment by meziantou** [comment]
[Meziantou.Analyzer@1.0.649](https://www.nuget.org/packages/Meziantou.Analyzer/1.0.649) should contain a fix for this issue
